shared/libraries/discovery/include/daq_discovery/mdnsdiscovery_client.h

```diff
@@ -327,67 +328,39 @@ inline void MDNSDiscoveryClient::openClientSockets(std::vector<int>& sockets, in
 #endif
 }
 
-inline mdns_string_t MDNSDiscoveryClient::ipv4AddressToString(
-    char* buffer, size_t capacity, const sockaddr_in* addr, size_t addrlen, bool includePort)
+inline std::string MDNSDiscoveryClient::ipv4AddressToString(const sockaddr_in* addr, size_t addrlen, bool includePort)
 {
     char host[NI_MAXHOST] = {0};
     char service[NI_MAXSERV] = {0};
-    int ret = getnameinfo(reinterpret_cast<const sockaddr*>(addr),
-                          static_cast<socklen_t>(addrlen),
-                          host,
-                          NI_MAXHOST,
-                          service,
-                          NI_MAXSERV,
-                          NI_NUMERICSERV | NI_NUMERICHOST);
-    int len = 0;
-    if (ret == 0)
-    {
-        if (addr->sin_port != 0 && includePort)
-            len = snprintf(buffer, capacity, "%s:%s", host, service);
-        else
-            len = snprintf(buffer, capacity, "%s", host);
-    }
-    if (len >= static_cast<int>(capacity))
-        len = static_cast<int>(capacity) - 1;
-    mdns_string_t str;
-    str.str = buffer;
-    str.length = len;
-    return str;
+    int ret = getnameinfo((const struct sockaddr*)addr, (socklen_t)addrlen, host, NI_MAXHOST,
+                        service, NI_MAXSERV, NI_NUMERICSERV | NI_NUMERICHOST);
+    if (ret != 0)
+        return "";
+
+    if (addr->sin_port != 0 && includePort)
+        return std::string(host) + ":" + service;
+    return std::string(host);
 }
 
-inline mdns_string_t MDNSDiscoveryClient::ipv6AddressToString(
-    char* buffer, size_t capacity, const sockaddr_in6* addr, size_t addrlen, bool includePort)
+inline std::string MDNSDiscoveryClient::ipv6AddressToString(const sockaddr_in6* addr, size_t addrlen, bool includePort)
 {
     char host[NI_MAXHOST] = {0};
     char service[NI_MAXSERV] = {0};
-    int ret = getnameinfo(reinterpret_cast<const sockaddr*>(addr),
-                          static_cast<socklen_t>(addrlen),
-                          host,
-                          NI_MAXHOST,
-                          service,
-                          NI_MAXSERV,
-                          NI_NUMERICSERV | NI_NUMERICHOST);
-    int len = 0;
-    if (ret == 0)
-    {
-        if (addr->sin6_port != 0 && includePort)
-            len = snprintf(buffer, capacity, "[%s]:%s", host, service);
-        else
-            len = snprintf(buffer, capacity, "%s", host);
-    }
-    if (len >= static_cast<int>(capacity))
-        len = static_cast<int>(capacity) - 1;
-    mdns_string_t str;
-    str.str = buffer;
-    str.length = len;
-    return str;
+    int ret = getnameinfo((const struct sockaddr*)addr, (socklen_t)addrlen, host, NI_MAXHOST,
+                          service, NI_MAXSERV, NI_NUMERICSERV | NI_NUMERICHOST);
+    if (ret != 0)
+        return "";
+
+    if (addr->sin6_port != 0 && includePort)
+        return "[" + std::string(host) + "]:" + service;
```

Comment:
```
   A host identified by an Internet Protocol literal address, version 6
   [[RFC3513](https://www.rfc-editor.org/rfc/rfc3513)] or later, is distinguished by enclosing the IP literal
   within square brackets ("[" and "]").  This is the only place where
   square bracket characters are allowed in the URI syntax.  In
   anticipation of future, as-yet-undefined IP literal address formats,
   an implementation may use an optional version flag to indicate such a
   format explicitly rather than rely on heuristic determination.

      IP-literal = "[" ( IPv6address / IPvFuture  ) "]"

      IPvFuture  = "v" 1*HEXDIG "." 1*( unreserved / sub-delims / ":" )
```

     [source](https://www.rfc-editor.org/rfc/rfc3986)